sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncClient.java

```diff
@@ -209,21 +217,7 @@ boolean isContentResponseOnWriteEnabled() {
      * an error.
      */
     public Mono<CosmosDatabaseResponse> createDatabaseIfNotExists(String id) {
-        return createDatabaseIfNotExistsInternal(getDatabase(id));
-    }
-
-    private Mono<CosmosDatabaseResponse> createDatabaseIfNotExistsInternal(CosmosAsyncDatabase database) {
-        return database.read().onErrorResume(exception -> {
-            final Throwable unwrappedException = Exceptions.unwrap(exception);
-            if (unwrappedException instanceof CosmosException) {
-                final CosmosException cosmosException = (CosmosException) unwrappedException;
-                if (cosmosException.getStatusCode() == HttpConstants.StatusCodes.NOTFOUND) {
-                    return createDatabase(new CosmosDatabaseProperties(database.getId()),
-                        new CosmosDatabaseRequestOptions());
-                }
-            }
-            return Mono.error(unwrappedException);
-        });
+        return withContext(context -> createDatabaseIfNotExistsInternal(getDatabase(id), null, context));
```

Comment:
We could do that, but avoid nesting call in public api. Also it would be easier to read and debug. And our tracer goal getting fulfill with current design, so if you don't mind, can we keep current design ?

Reply:
@kushagraThapar Could you clarify if this is a design decision to not reuse the methods here?
And if it is, why is it followed in some files and not all the time?
More context - https://github.com/Azure/azure-sdk-for-java/pull/12867#discussion_r452564457

Reply:
Sure we can, but we should not touch public surface implementations here as this PR should only target the tracing APIs. 
I have created a work item for this refactor work : https://github.com/Azure/azure-sdk-for-java/issues/13031